_1327/documents/utils.py

```diff
@@ -61,6 +63,12 @@ def handle_edit(request, document, formset=None, initial=None):
 			if not document.has_perms():
 				document.set_all_permissions(cleaned_data['group'])
 
+			if content_type == ContentType.objects.get_for_model(Poll):
```

Comment:
how about adding a new function for each document that can be called to add additional perms while editing?
This should do nothing in the base class and can be overriden in each subclass